TestFramework.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author Michael Ray
+ * @since 3/22/2017
+ * @version 1.0.0
+ * 
+ */
+#ifndef TESTFRAMEWORK_H
+#define TESTFRAMEWORK_H
+
+#include "cmsis.h"
+#include "pinmap.h"
+#include "PeripheralPins.h"
+#include "mbed.h"
+
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "ci_test_config.h"
+
+#include <vector>
+#include <cmath>
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#define TEST_STRING_MAX 100
+
+class TestFramework {
+public:
+
+	enum Type {
+		AnalogInput,
+		AnalogOutput,
+		DigitalIO,
+		PWM,
+		I2C_SDA,
+		I2C_SCL,
+		SPI_CLK,
+		SPI_MISO,
+		SPI_MOSI,
+		SPI_CS,
+		BusIO,
+		CITS_AnalogInput,
+		CITS_AnalogOutput,
+		CITS_DigitalIO,
+		CITS_PWM,
+		CITS_I2C_SDA,
+		CITS_I2C_SCL,
+		CITS_SPI_MOSI,
+		CITS_SPI_MISO,
+		CITS_SPI_CLK,
+		CITS_SPI_CS,
+		TS_NC
+	};
+
+	static std::vector< std::vector <PinMap> > pinout;
+	static std::vector<unsigned int> pin_iterators;
+
+	TestFramework();
+
+	static bool check_size(Type pintype);
```

Comment:
Yes, it is. The header file is for declarations, I believe it is stated in our coding standard. The header file provides a declaration, implementation follows it (not the other way around).